backend/eventloop/application_event_loop/application_event_runner_deployments.go

```diff
@@ -734,6 +734,20 @@ func (a *applicationEventLoopRunner_Action) applicationEventRunner_handleUpdateD
 		return err
 	}
 
+	var comparedTo fauxargocd.FauxComparedTo
+	comparedTo, err = retrieveComparedToFieldInApplicationState(applicationState.ReconciledState)
+	if err != nil {
+		a.log.Error(err, "unable to retrieve comparedTo field in ApplicationState")
+		return err
+	}
+
+	// Update gitopsDeployment status with reconciledState
+	gitopsDeployment.Status.ReconciledState.Source.Path = comparedTo.Source.Path
+	gitopsDeployment.Status.ReconciledState.Source.RepoURL = comparedTo.Source.RepoURL
+	gitopsDeployment.Status.ReconciledState.Source.Branch = comparedTo.Source.TargetRevision
+	gitopsDeployment.Status.ReconciledState.Destination.Name = comparedTo.Destination.Name
```

Comment:
When I try out the PR, and I view the `reconciledState` in the `GitOpsDeployment` resource, it looks like this:

    reconciledState:
      destination:
        name: 5c59aecf-6ae1-4801-b52a-4641f464e7ef <--- the managedenvironment_id of the ManagedEnviroment row
        namespace: jane
      source:
        branch: ""
        path: environments/overlays/dev
        repoURL: ""


Rather than displaying the database primary key for the user (we generally want to keep our database internals hidden), we should instead convert this "5c59aecf-(...)" value into a reference to the name of the corresponding GitOpsDeploymentManagedEnvironment CR in the environment. (This is included as part of the story criteria).
- For example, let's say we have a GitOpsDeploymentManagedEnvironment name `my-env`, defined in the namespace.
- That GDManagedEnvironment resource corresponds to the `5c59aecf-6ae1-4801-b52a-4641f464e7ef` ManagedEnvironment row.
- Therefore, in .reconciledState.destination.name, we should use the value `my-env`

In order to get the name of the GitOpsDeploymentManagedEnvironment resource corresponding to the ManagedEnvironment table, we can use the `APICRToDatabaseMapping` table. It is the responsibility of this table to maintain the relationship from K8s API resources (like GitOpsDeploymentManagedEnvironment) <-> corresponding database resources (row in ManagedEnvironment table).

I checked, and it looks like we don't have an existing database query function for this, so I wrote one (which hopefully works, I've not tested it).

You can add this to apicrtodatabasemapping.go (you'll also need to add it to the `ApplicationScopedQueries` interface in `queries.go`)

```go
// GetAPICRForDatabaseUID retrieves the name/namespace/uid of an API Resources (such as GitOpsDeploymentManagedEnvironment) based on the
// primary key of the corresponding database row (for example, ManagedEnvironment)
func (dbq *PostgreSQLDatabaseQueries) GetAPICRForDatabaseUID(ctx context.Context, obj *APICRToDatabaseMapping) error {

	if err := validateQueryParamsEntity(obj, dbq); err != nil {
		return err
	}

	if err := isEmptyValues("GetAPICRForDatabaseUID",
		"APIResourceType", obj.APIResourceType,
		"DBRelationType", obj.DBRelationType,
		"DBRelationKey", obj.DBRelationKey); err != nil {
		return err
	}

	var result []APICRToDatabaseMapping

	if err := dbq.dbConnection.Model(&result).
		// TODO: GITOPSRVCE-68 - PERF - Add a DB index for this
		Where("atdbm.api_resource_type = ?", obj.APIResourceType).
		Where("atdbm.db_relation_type = ?", obj.DBRelationType).
		Where("atdbm.db_relation_key = ?", obj.DBRelationKey).
		Context(ctx).
		Select(); err != nil {

		return fmt.Errorf("error on retrieving database mapping for APICRToDatabase: %v", err)
	}

	if len(result) == 0 {
		return NewResultNotFoundError(fmt.Sprintf("unable to retrieve APICRToDatabase mapping for %s:%s",
			obj.APIResourceType, obj.DBRelationKey))
	}

	if len(result) > 1 {
		return fmt.Errorf("unexpected number of results when retrieving APICRToDatabase mapping for %s:%s",
			obj.APIResourceType, obj.DBRelationKey)
	}

	*obj = result[0]

	return nil

}
``

Then, in backend, when you read the value of the managedEnvironment ID from the ReconciledState field of the ApplicationState database table, you can convert the value of this field to the corresponding 
- You can use the above function to translate the database ID -> CR name
- If no match is found (for example, because the managedenvironment was deleted), just leave the field empty (rather than returning an error)